svc/api/admin/src/route/clusters/datacenters.rs

```diff
@@ -147,3 +147,47 @@ pub async fn taint(
 
 	Ok(json!({}))
 }
+
+// MARK: PUT /admin/clusters/{cluster_id}/datacenters/{datacenter_id}
+pub async fn update(
+	ctx: Ctx<Auth>,
+	cluster_id: Uuid,
+	datacenter_id: Uuid,
+	body: models::AdminClustersDatacentersUpdateRequest,
+) -> GlobalResult<Value> {
+	// Make sure that the datacenter is part of the cluster
+	let datacenters = op!([ctx] cluster_datacenter_get {
+		datacenter_ids: vec![datacenter_id.into()],
+	})
+	.await?
+	.datacenters;
+
+	let datacenter = unwrap!(datacenters.first(), "datacenter does not exist");
+
+	if datacenter.cluster_id != Some(cluster_id.into()) {
+		bail!("datacenter does not belong to cluster");
+	}
+
+	let pools = vec![cluster::msg::datacenter_update::PoolUpdate {
+		pool_type: backend::cluster::PoolType::api_from(body.pool_type) as i32,
+		hardware: body
+			.hardware
+			.iter()
+			.map(|h| backend::cluster::Hardware {
+				provider_hardware: h.provider_hardware.clone(),
+			})
+			.collect(),
+		desired_count: body.desired_count.map(|c| c as u32),
+		max_count: body.max_count.map(|c| c as u32),
+		drain_timeout: body.drain_timeout.map(|d| d as u64),
+	}];
```

Comment:
this will replace _all_ pools instead of only update the requested pools

right @MasterPtato ?

Reply:
@NathanFlurry no, the code in `cluster-datacenter-update` only patches the pools and does not overwrite them